### PR TITLE
[aten] fix vector memory leak

### PR DESCRIPTION
**Pull request by gaurav1086**
free(y) missing.

### CODE REVIEW

File: aten/src/TH/vector/VSX.cpp [hunk 1]
```diff
@@ -2450,6 +2461,7 @@ void test_THFloatVector_divs_VSX()
     free(y_standard);
     free(y_optimized);
     free(x);
+    free(y);
```
Comment (agolynski) [inline aten/src/TH/vector/VSX.cpp]: is y defined and allocated in this function?

Comment (gaurav1086) [reply aten/src/TH/vector/VSX.cpp]: Thanks for the review. Corrected.

Comment (gaurav1086) [reply aten/src/TH/vector/VSX.cpp]: I'm surprised as to how it compiled.

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 17d6fbaf:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=17d6fbaf2d67edc13daa30a922c0791e943bd10a).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4323937) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=89859415) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=89859415))
<details>
<summary>
<code>Jan 23 17:12:19 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-device_test.diff</code>
</summary>

```
Jan 23 17:12:19 File to patch:  
Jan 23 17:12:19 Skip this patch? [y]  
Jan 23 17:12:19 4 out of 4 hunks ignored 
Jan 23 17:12:19 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 23 17:12:19 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-device_test.diff 
Jan 23 17:12:19 Traceback (most recent call last): 
Jan 23 17:12:19   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Jan 23 17:12:19     patch_repo(args) 
Jan 23 17:12:19   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Jan 23 17:12:19     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Jan 23 17:12:19 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-device_test.diff 
Jan 23 17:12:19 =================== sccache compilation log =================== 
Jan 23 17:12:19 + cleanup 
Jan 23 17:12:19 + retcode=1 
Jan 23 17:12:19 + set +x 
Jan 23 17:12:19 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 23 17:12:19 Compile requests                 1 
Jan 23 17:12:19 Compile requests executed        0 
Jan 23 17:12:19 Cache hits                       0 
Jan 23 17:12:19 Cache misses                     0 
Jan 23 17:12:19 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Review by agolynski — APPROVED** [review]
Thank you for the fix!

**Comment by agolynski** [comment]
Seems like there could be some conflict on xla as I've seen on test failures:

Jan 23 17:12:19 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff
Jan 23 17:12:19 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-device_test.diff
Jan 23 17:12:19 Traceback (most recent call last):
Jan 23 17:12:19   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module>
Jan 23 17:12:19     patch_repo(args)
Jan 23 17:12:19   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo
Jan 23 17:12:19     raise RuntimeError('Failed to apply patch: {}'.format(ppath))
Jan 23 17:12:19 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-device_test.diff

do you mind rebasing on master and we'll rerun the tests?

**Comment by gaurav1086** [comment]
Rebased with latest master and pushed.

**Comment by gaurav1086** [comment]
@agolynski , rebased with the latest master.

**Comment by gaurav1086** [comment]
Hi, is there anything else required from my side ? Thank you.

**Comment by gaurav1086** [comment]
@agolynski , looks like the build failure is unrelated to my change.

**Comment by agolynski** [comment]
Sorry for the delay. I agree, seems like the failures seems to be unrelated.

**Comment by gaurav1086** [comment]
Cool, thank you. Kindly let me know if there are any other changes needed in this fix.

**Comment by ezyang** [comment]
@agolynski you're going to merge this right? Let me know if you need help.

**Pull request closed without merging**